lib/mpl_toolkits/basemap/__init__.py

```diff
@@ -4777,10 +4777,15 @@ def shiftdata(self,lonsin,datain=None,lon_0=None):
             lonsin1 = lonsin[0,:]
             lonsin1 = np.where(lonsin1 > lon_0+180, lonsin1-360 ,lonsin1)
             lonsin1 = np.where(lonsin1 < lon_0-180, lonsin1+360 ,lonsin1)
-            londiff = np.abs(lonsin1[0:-1]-lonsin1[1:])
-            londiff_sort = np.sort(londiff)
-            thresh = 360.-londiff_sort[-2]
-            itemindex = nlons-np.where(londiff>=thresh)[0]
+            if nlons > 1:
+                londiff = np.abs(lonsin1[0:-1]-lonsin1[1:])
+                londiff_sort = np.sort(londiff)
+                thresh = 360.-londiff_sort[-2] if nlons > 2 else 360.-londiff_sort[-1]
+                itemindex = nlons-np.where(londiff>=thresh)[0]
+            else:
+                lonsin[0, :] = lonsin1
```

Comment:
It shifts one point to the projection region.

In 2d case lonsin is copied to lonsin1 and then lonsin1 is shifted.. Then
lonsin is also shifted below if itemindex is not 0. But when there is only
one point itemindex is zero, so I had to copy back the shifted lonsin1...

2015-10-26 15:51 GMT-04:00 Benjamin Root <notifications@github.com>:

> In lib/mpl_toolkits/basemap/__init__.py
> <https://github.com/matplotlib/basemap/pull/215#discussion_r43043356>:
>
> > @@ -4777,10 +4777,15 @@ def shiftdata(self,lonsin,datain=None,lon_0=None):
> >              lonsin1 = lonsin[0,:]
> >              lonsin1 = np.where(lonsin1 > lon_0+180, lonsin1-360 ,lonsin1)
> >              lonsin1 = np.where(lonsin1 < lon_0-180, lonsin1+360 ,lonsin1)
> > -            londiff = np.abs(lonsin1[0:-1]-lonsin1[1:])
> > -            londiff_sort = np.sort(londiff)
> > -            thresh = 360.-londiff_sort[-2]
> > -            itemindex = nlons-np.where(londiff>=thresh)[0]
> > +            if nlons > 1:
> > +                londiff = np.abs(lonsin1[0:-1]-lonsin1[1:])
> > +                londiff_sort = np.sort(londiff)
> > +                thresh = 360.-londiff_sort[-2] if nlons > 2 else 360.-londiff_sort[-1]
> > +                itemindex = nlons-np.where(londiff>=thresh)[0]
> > +            else:
> > +                lonsin[0, :] = lonsin1
>
> What's the point of this line?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/basemap/pull/215/files#r43043356>.
>



-- 
Sasha